pydm/widgets/timeplot.py

```diff
@@ -231,23 +529,106 @@ def setCurves(self, new_list):
+
+    def refreshCurve(self, curve):
+        """
+        Remove a curve currently being plotted on the timeplot, then redraw that curve, which could have been updated
+        with a new symbol, line style, line width, etc.
+
+        Parameters
+        ----------
+        curve : TimePlotCurveItem
+            The curve to be re-added.
+        """
+        curve = self.findCurve(curve.channel)
+        if curve:
+            self.removeYChannel(curve)
+            self.addYChannel(y_channel=curve.address, color=curve.color, name=curve.address,
+                             lineStyle=curve.lineStyle, lineWidth=curve.lineWidth, symbol=curve.symbol,
+                             symbolSize=curve.symbolSize)
+
+    def addLegendItem(self, item, pv_name, force_show_legend=False):
+        """
+        Add an item into the graph's legend.
+
+        Parameters
+        ----------
+        item : TimePlotCurveItem
+            A curve being plotted in the graph
+        pv_name : str
+            The PV channel
+        force_show_legend : bool
+            True to make the legend to be displayed; False to just add the item, but do not display the legend.
+        """
+        self._legend.addItem(item, pv_name)
+        self.setShowLegend(force_show_legend)
+
+    def removeLegendItem(self, pv_name):
+        """
+        Remove an item from the legend.
+
+        Parameters
+        ----------
+        pv_name : str
+            The PV channel, used to search for the legend item to remove.
+        """
+        self._legend.removeItem(pv_name)
+        if len(self._legend.items) == 0:
+            self.setShowLegend(False)
+
     def getBufferSize(self):
+        """
+        Get the size of the data buffer for the entire chart.
+
+        Returns : int
+        -------
+        The chart's data buffer size.
+        """
         return int(self._bufferSize)
 
     def setBufferSize(self, value):
+        """
+        Set the size of the data buffer of the entire chart. This will also update the same value for each of the
+        data buffer of each chart's curve.
+
+        Parameters
+        ----------
+        value : int
+            The new buffer size for the chart.
+        """
         if self._bufferSize != int(value):
-            self._bufferSize = max(int(value), 1)
+            # Originally, the bufferSize is the max between the user's input and 1, and 1 doesn't make sense.
+            # So, I'm comparing the user's input with the minimum buffer size, and pick the max between the two
+            self._bufferSize = max(int(value), MINIMUM_BUFFER_SIZE)
             for curve in self._curves:
                 curve.setBufferSize(value)
 
     def resetBufferSize(self):
-        if self._bufferSize != 1200:
-            self._bufferSize = 1200
+        """
+        Reset the data buffer size of the chart, and each of the chart's curve's data buffer, to the minimum
+        """
+        if self._bufferSize != MINIMUM_BUFFER_SIZE:
```

Comment:
As mentioned in an earlier comment, I think this should reset the buffer size to a new `DEFAULT_BUFFER_SIZE` constant, not the minimum buffer size.

Reply:
Done.